### PR TITLE
p2p spec updates

### PR DESCRIPTION
**Pull request by 0xSulpiride**
## Description

- feat: https://github.com/eth-infinitism/bundler-spec/pull/21
- feat: https://github.com/eth-infinitism/bundler-spec/pull/18
- feat: https://github.com/eth-infinitism/bundler-spec/pull/17
- feat: https://github.com/eth-infinitism/bundler-spec/pull/15

## Types of changes
What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Other (please describe): 

## Further comments (optional)

-

### CODE REVIEW

**Comment by 0xSulpiride** [comment]
We need to agree on the size of MempoolId. The IPFS hash is 46 bytes, so I'd push for this PR - https://github.com/eth-infinitism/bundler-spec/pull/19

**Comment by 0xSulpiride** [comment]
Also we need to switch to one bundler - one chain model - that will be a separate PR

File: packages/node/src/network/metadata.ts [hunk 1]
```diff
@@ -23,35 +19,27 @@ export enum SubnetType {
 export class MetadataController {
   private setEnrValue?: (key: string, value: Uint8Array) => Promise<void>;
   private _metadata: ts.Metadata;
+  private _chainId: number;
```
Comment (ch4r10t33r) [inline packages/node/src/network/metadata.ts:22]: Any reason for the _ prefix?

File: packages/node/src/network/network.ts [hunk 2]
```diff
@@ -233,8 +238,8 @@ export class Network implements INetwork {
 
   async getMetadata(): Promise<ts.Metadata> {
     return {
-      seqNumber: this.metadata.seqNumber,
-      mempoolSubnets: this.metadata.mempoolSubnets,
+      seq_number: this.metadata.seq_number,
```
Comment (ch4r10t33r) [inline packages/node/src/network/network.ts:241]: Have the variable names changed in the spec?

File: packages/executor/src/modules/eth.ts [hunk 3]
```diff
@@ -105,17 +106,21 @@ export class Eth {
...
+      if (nodeApi) {
+        const { mempoolId } = getCanonicalMempool(this.chainId, {
+          entryPoint: this.config.canonicalEntryPoint,
+          mempoolId: this.config.canonicalMempoolId,
+        });
+        if (mempoolId.length > 0) {
+          const blockNumber = await this.provider.getBlockNumber(); // TODO: fetch blockNumber from simulateValidation
+          await nodeApi.publishVerifiedUserOperationJSON(
+            entryPoint,
+            userOp,
```
[the remark this answers is not in the document]
Comment (0xSulpiride) [reply packages/executor/src/modules/eth.ts:118]: It's respective to the specs - https://github.com/eth-infinitism/bundler-spec/blob/main/p2p-specs/p2p-interface.md#verifieduseroperation

File: packages/node/src/network/metadata.ts (hunk 1 above)
Comment (0xSulpiride) [reply packages/node/src/network/metadata.ts]: nope, will delete that

**Pull request merged**